build/spot.js

```diff
@@ -2725,6 +2725,86 @@ function addPhiWrapped(candidates, options){
 }
 
 })();
+
+function containsClitic(x){
+	return x.indexOf("clitic")>-1;
+}
+
+function generateWordOrders(wordList, clitic){
+	if(typeof wordList === 'string'){
+		var cliticTagIndex = wordList.indexOf("-clitic");
+		if(cliticTagIndex > 0){
+			var wordListParts = wordList.split("-clitic");
+			wordList = wordListParts[0]+wordListParts[1];
+		}
+		wordList = wordList.split(' ');
+	}
+	//Find the clitic to move around
+	var cliticIndex = wordList.indexOf(clitic);
+	if(cliticIndex < 0)
+		throw new Error("The provided clitic"+clitic+" was not found in the word list");
+	//Slice the clitic out
+	var beforeClitic = wordList.slice(0,cliticIndex);
+	var afterClitic = wordList.slice(cliticIndex+1, wordList.length);
+	var cliticlessWords = beforeClitic.concat(afterClitic);
+
+	var orders = new Array(wordList.length);
+	for(var i = 0; i < wordList.length; i++){
```

Comment:
Looks like the case where the clitic is the last item is not generated? Is this intentional?

Reply:
It is generated because wordList.length = cliticlessWords.length +1. But I changed this for readability.